### PR TITLE
WebGLRenderer: Refactored vertex color alpha code

### PR DESCRIPTION
**Pull request by mrdoob**
Related issue: #20975

**Description**

Refactored some of the logic and glsl defines in the vertex color alpha code.

@chubei @Mugen87 What do you guys think?

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1524,10 +1522,6 @@ function WebGLRenderer( parameters ) {
 
 				needsProgramChange = true;
 
-			} else if ( materialProperties.vertexAlpha !== vertexAlpha ) {
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js:1527]: I'm not sure why is this needed for...

File: src/renderers/shaders/ShaderChunk/color_pars_vertex.glsl.js [hunk 2]
```diff
@@ -1,15 +1,11 @@
 export default /* glsl */`
-#if defined( USE_COLOR ) || defined( USE_INSTANCING_COLOR )
+#if defined( USE_COLOR_ALPHA )
 
-	#if defined( USE_VERTEX_ALPHA )
+	varying vec4 vColor;
 
-		varying vec4 vColor;
+#elif defined( USE_COLOR ) || defined( USE_INSTANCING_COLOR )
```
Comment (mrdoob) [inline src/renderers/shaders/ShaderChunk/color_pars_vertex.glsl.js:6]: This is kind of ugly. Sure needs another clean up pass in the future.

File: src/renderers/webgl/WebGLPrograms.js [hunk 3]
```diff
@@ -208,7 +208,7 @@ function WebGLPrograms( renderer, cubemaps, extensions, capabilities, bindingSta
 
 			vertexTangents: ( material.normalMap && material.vertexTangents ),
 			vertexColors: material.vertexColors,
-			vertexAlpha: ( ( object.isMesh || object.isLine || object.isPoints ) && object.geometry.attributes.color && object.geometry.attributes.color.itemSize === 4 ),
+			vertexAlphas: material.vertexColors && object.geometry.attributes.color && object.geometry.attributes.color.itemSize === 4,
```
Comment (mrdoob) [inline src/renderers/webgl/WebGLPrograms.js:211]: We are not deprecating `material.vertexColors` yet so the right way to do it to me.

File: src/renderers/WebGLRenderer.js (hunk 1 above)
Comment (Mugen87) [reply src/renderers/WebGLRenderer.js:1527]: These is required otherwise it won't work if a material is shared between a geometry with vertex colors and without.

File: src/renderers/webgl/WebGLPrograms.js (hunk 3 above)
Comment (Mugen87) [reply src/renderers/webgl/WebGLPrograms.js:211]: This looks good!

File: src/renderers/WebGLRenderer.js (hunk 1 above)
Comment (Mugen87) [reply src/renderers/WebGLRenderer.js:1527]: Some background information: https://github.com/mrdoob/three.js/pull/20975#discussion_r588901717

Comment (mrdoob) [reply src/renderers/WebGLRenderer.js:1527]: Okay, I follow. Thanks!

File: src/renderers/WebGLRenderer.js [hunk 4]
```diff
@@ -1466,7 +1466,7 @@ function WebGLRenderer( parameters ) {
 		const environment = material.isMeshStandardMaterial ? scene.environment : null;
 		const encoding = ( _currentRenderTarget === null ) ? _this.outputEncoding : _currentRenderTarget.texture.encoding;
 		const envMap = cubemaps.get( material.envMap || environment );
-		const vertexAlpha = ( ( object.isMesh || object.isLine || object.isPoints ) && object.geometry.attributes.color && object.geometry.attributes.color.itemSize === 4 );
+		const vertexAlphas = material.vertexColors === true && object.geometry.attributes.color && object.geometry.attributes.color.itemSize === 4;
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js:1469]: @Mugen87 This should work then, right?

Comment (Mugen87) [reply src/renderers/WebGLRenderer.js:1469]: Looks good!
